pandas/_libs/tslibs/period.pyx

```diff
@@ -1651,7 +1654,7 @@ cdef class _Period(PeriodMixin):
             return NaT
         else:
             freq = cls._maybe_convert_freq(freq)
-            self = _Period.__new__(cls, ordinal, freq)
+            self = _Period.__new__(cls, ordinal=ordinal, freq=freq)
```

Comment:
Looks like this can be undone

Reply:
Agreed, this is just some additional safety.
I wrote this when I noticed that `Period.__new__` (and not `_Period.__new__`) has a totally different parameter order. The problem is that this particular line is in a method on `Period` not on `_Period`. So if you read a bit too fast this is very misleading - I thought this was the cause of the error on numpy dev runner :)

Anyway, I'll remove if you prefer.

Reply:
Happy to have this change in another PR instead to keep the scoping of this PR tight

Reply:
ok